ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplLoadTask.java

```diff
@@ -554,6 +554,24 @@ public void run() throws SemanticException {
           }
         }
       });
+      if (work.shouldFailover()) {
+        listOfPreAckTasks.add(new PreAckTask() {
+          @Override
+          public void run() throws SemanticException {
+            try {
+              Database db = getHive().getDatabase(work.getTargetDatabase());
+              if (!MetaStoreUtils.isTargetOfReplication(db)) {
+                return;
+              }
+              LOG.info("Removing property: {} from database: {}", ReplConst.TARGET_OF_REPLICATION, db.getName());
+              db.getParameters().remove(ReplConst.TARGET_OF_REPLICATION);
```

Comment:
If you remove the property TARGET_OF_REPLICATION at this point of time which will allow the background threads like  Partition Management/ Stat Updater etc. Can we still rollback in that case? Wondering if we should delay the removal of this property until first dump in reverse direction?

Reply:
Yes, rollback may be impacted. But we can't delay this removal until first dump also. As dump operation will itself fail if db is target of replication.

Reply:
We can remove this property as the first dump in reverse direction